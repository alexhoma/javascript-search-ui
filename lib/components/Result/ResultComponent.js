"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
exports.__esModule = true;
var ItemUUID_1 = require("apisearch/lib/Model/ItemUUID");
var preact_1 = require("preact");
var compat_1 = require("preact/compat");
var Constants_1 = require("../../Constants");
var Container_1 = require("../../Container");
var Template_1 = require("../Template");
var defaultTemplates_1 = require("./defaultTemplates");
var Item_1 = require("./Item");
var ResultActions_1 = require("./ResultActions");
/**
 * Result Component
 */
var ResultComponent = /** @class */ (function (_super) {
    __extends(ResultComponent, _super);
    /**
     * Constructor
     */
    function ResultComponent(props) {
        var _this = _super.call(this, props) || this;
        _this.fromLoadingNextPage = false;
        _this.observer = compat_1.useRef();
        _this.endResultsBoxRef = compat_1.useCallback(function (node) {
            if (_this.observer.current instanceof IntersectionObserver) {
                _this.observer.current.disconnect();
            }
            _this.observer.current = new IntersectionObserver(function (entries) {
                if (entries[0].isIntersecting) {
                    var _a = _this.props, environmentId = _a.environmentId, store = _a.store, repository = _a.repository;
                    _this.fromLoadingNextPage = true;
                    ResultActions_1.infiniteScrollNextPageAction(environmentId, store.getCurrentQuery(), repository, _this.state.page + 1);
                }
            });
            if ((_this.observer.current instanceof IntersectionObserver) && node) {
                _this.observer.current.observe(node);
            }
        }, []);
        _this.state = {
            items: [],
            page: 0,
            hasNewPage: false,
            focus: props.fadeInSelector === ""
        };
        return _this;
    }
    /**
     * Hook that change state once mouse clicks inside or outside the container
     */
    ResultComponent.prototype.addMouseDownListeners = function (ref, fadeInSelector) {
        var _this = this;
        compat_1.useEffect(function () {
            var self = _this;
            /**
             * Alert if clicked on outside of element
             */
            function handleClickOutside(event) {
                self.setState(function () {
                    return {
                        focus: event.target.closest(fadeInSelector) != null
                    };
                });
            }
            // Bind the event listener
            document.addEventListener("mousedown", handleClickOutside);
            return function () {
                // Unbind the event listener on clean up
                document.removeEventListener("mousedown", handleClickOutside);
            };
        }, [ref]);
    };
    /**
     * Component will receive props
     *
     * @param props
     */
    ResultComponent.prototype.componentWillReceiveProps = function (props) {
        if (props.store.getCurrentResult() == null) {
            this.setState(function (prevState) {
                return {
                    items: [],
                    page: 0,
                    hasNewPage: false
                };
            });
            return;
        }
        var currentResult = props.store.getCurrentResult();
        var currentQuery = props.store.getCurrentQuery();
        var items = currentResult.getItems();
        var currentPage = currentQuery.getPage();
        var hasNewPage = (currentResult.getTotalHits() > (currentPage * currentQuery.getSize()));
        var currentItems = this.fromLoadingNextPage
            ? this.state.items.concat(items)
            : items;
        this.fromLoadingNextPage = false;
        this.setState(function (prevState) {
            return {
                items: currentItems,
                page: props.store.getCurrentQuery().getPage(),
                hasNewPage: hasNewPage
            };
        });
    };
    /**
     * Component will mount
     */
    ResultComponent.prototype.componentWillMount = function () {
        var props = this.props;
        /**
         * Dispatch action
         */
        ResultActions_1.configureQuery(props.environmentId, props.store.getCurrentQuery(), props.itemsPerPage, props.highlightsEnabled, props.promote.map(function (itemUUID) {
            return itemUUID instanceof ItemUUID_1.ItemUUID
                ? itemUUID
                : ItemUUID_1.ItemUUID.createFromArray(itemUUID);
        }), props.exclude.map(function (itemUUID) {
            return itemUUID instanceof ItemUUID_1.ItemUUID
                ? itemUUID
                : ItemUUID_1.ItemUUID.createFromArray(itemUUID);
        }), props.fields, props.filter, props.minScore);
    };
    /**
     * Render
     *
     * @return {any}
     */
    ResultComponent.prototype.render = function () {
        var _this = this;
        var _a;
        var props = this.props;
        var dirty = props.store.isDirty();
        var containerClassName = props.classNames.container;
        var itemsListClassName = props.classNames.itemsList;
        var placeholderClassName = props.classNames.placeholder;
        var itemsListTemplate = props.template.itemsList;
        var placeholderTemplate = (_a = props.template.placeholder) !== null && _a !== void 0 ? _a : "";
        var currentResult = props.store.getCurrentResult();
        var currentQuery = props.store.getCurrentQuery();
        var currentVisibleResults = props.currentVisibleResults;
        var wrapperRef = compat_1.useRef(null);
        var hasInfiniteScrollNextPage = (props.infiniteScroll !== false) &&
            ((props.infiniteScroll === true) ||
                (props.infiniteScroll >= 0)) &&
            this.state.hasNewPage;
        var infiniteScrollMargin = hasInfiniteScrollNextPage
            ? (props.infiniteScroll === true
                ? 0
                : props.infiniteScroll)
            : undefined;
        if (props.fadeInSelector !== "") {
            this.addMouseDownListeners(wrapperRef, props.fadeInSelector);
        }
        if (!currentVisibleResults || !this.state.focus) {
            return (preact_1.h("div", { className: "as-result " + containerClassName }));
        }
        /**
         * Data accessible to the template
         */
        var items = this.state.items;
        var reducedTemplateData = {
            query: currentQuery.getQueryText(),
            suggestions: currentResult.getSuggestions()
        };
        /**
         * We should add positions to items
         * When the number of items to render is higher than the page size, we are in front of infinite scroll
         */
        var isInfinite = (currentQuery.getPage() === 1) || items.length > currentQuery.getSize();
        var firstItem = ((currentQuery.getPage() - 1) * currentQuery.getSize());
        var itemsForEvent = items;
        if (isInfinite) {
            itemsForEvent = Array.prototype.slice.call(items, firstItem);
        }
        Array.prototype.forEach.call(itemsForEvent, function (item) {
            item.position = ++firstItem;
            item.id = item.getId();
        });
        window.postMessage({
            name: "apisearch_result_items",
            query: currentQuery.toArray(),
            query_text: currentQuery.getQueryText(),
            with_results: items.length > 0,
            page: currentQuery.getPage(),
            site: props.store.getSite(),
            device: props.store.getDevice(),
            items: itemsForEvent
        }, "*");
        /**
         * Uses defined a custom items list. Old version
         */
        if (props.template.itemsList !== defaultTemplates_1.defaultItemsListTemplate) {
            return (preact_1.h("div", { className: "as-result " + containerClassName, ref: wrapperRef, style: "position: relative" },
                (dirty)
                    ? preact_1.h(Template_1["default"], { template: placeholderTemplate, className: "as-result__placeholder " + placeholderClassName, dictionary: this.props.dictionary })
                    : preact_1.h(Template_1["default"], { template: itemsListTemplate, data: __assign(__assign({}, reducedTemplateData), { items: (items)
                                ? items.map(function (item) { return _this.hydrateItem(item); })
                                : [] }), className: "as-result__itemsList " + itemsListClassName, dictionary: this.props.dictionary }),
                hasInfiniteScrollNextPage
                    ? preact_1.h("div", { ref: this.endResultsBoxRef, style: "bottom: " + infiniteScrollMargin + "px; position: relative;" })
                    : ""));
        }
        /**
         * New version
         */
        return (preact_1.h("div", { className: "as-result " + containerClassName, ref: wrapperRef }, (dirty)
            ? preact_1.h(Template_1["default"], { template: placeholderTemplate, className: "as-result__placeholder " + placeholderClassName, dictionary: this.props.dictionary })
            : ((items.length > 0)
                ? (preact_1.h("div", { className: "as-result__itemsList " + props.classNames.itemsList },
                    items.map(function (item) {
                        return preact_1.h(Item_1["default"], { data: __assign(__assign({}, reducedTemplateData), _this.hydrateItem(item)), template: props.template.item, className: "as-result__item " + props.classNames.item, dictionary: _this.props.dictionary });
                    }),
                    hasInfiniteScrollNextPage
                        ? preact_1.h("div", { id: "as-result__infinite_scroll_inspector", ref: this.endResultsBoxRef, style: "bottom: " + infiniteScrollMargin + "px; position: relative; width: 100%;" })
                        : ""))
                : preact_1.h(Template_1["default"], { template: props.template.noResults, data: {
                        query: currentQuery.getQueryText()
                    }, className: "as-result__noresults " + props.classNames.noResults, dictionary: this.props.dictionary }))));
    };
    /**
     * @param item
     */
    ResultComponent.prototype.hydrateItem = function (item) {
        var props = this.props;
        var environmentId = props.environmentId;
        var config = Container_1["default"].get(Constants_1.APISEARCH_CONFIG + "__" + environmentId);
        var apisearchUI = Container_1["default"].get(Constants_1.APISEARCH_UI + "__" + environmentId);
        var apisearchReference = apisearchUI.reference;
        var appId = config.app_id;
        var appUUID = item.getAppUUID();
        if (typeof appUUID === "object") {
            appId = appUUID.composedUUID();
        }
        var indexId = config.index_id;
        var indexUUID = item.getIndexUUID();
        if (typeof indexUUID === "object") {
            indexId = indexUUID.composedUUID();
        }
        var itemId = item.getUUID().composedUUID();
        var mainFields = {};
        Object.assign(mainFields, item.getMetadata(), item.getIndexedMetadata());
        var fieldsConciliation = {};
        Object.keys(props.fieldsConciliation).map(function (field, index) {
            var _a;
            fieldsConciliation[field] = (_a = mainFields[props.fieldsConciliation[field]]) !== null && _a !== void 0 ? _a : undefined;
        });
        Object.assign(mainFields, fieldsConciliation);
        item.fields = mainFields;
        var queryText = "";
        if (this.props.store.getCurrentQuery()) {
            queryText = this.props.store.getCurrentQuery().getQueryText();
        }
        return __assign(__assign({}, props.formatData(item)), {
            key: "item_" + itemId,
            uuid_composed: itemId,
            click: apisearchReference + '.click("' + appId + '", "' + indexId + '", "' + itemId + '");',
            add_to_cart: apisearchReference + '.interact("add_cart", "' + appId + '", "' + indexId + '", "' + itemId + '");',
            query_text: queryText,
            highlights_enabled: this.props.highlightsEnabled,
            striptags: function () {
                return function (val, render) { return render(val).replace(/(<([^>]+)>)/ig, ""); };
            }
        });
    };
    return ResultComponent;
}(preact_1.Component));
ResultComponent.defaultProps = {
    fields: [],
    itemsPerPage: 10,
    highlightsEnabled: false,
    promote: [],
    exclude: [],
    filter: function (query) { return null; },
    classNames: {
        container: "",
        itemsList: "",
        item: "",
        noResults: "",
        placeholder: ""
    },
    template: {
        itemsList: defaultTemplates_1.defaultItemsListTemplate,
        item: defaultTemplates_1.defaultItemTemplate,
        noResults: defaultTemplates_1.defaultNoResultsItemTemplate,
        placeholder: null
    },
    formatData: function (data) { return data; },
    fadeInSelector: "",
    fieldsConciliation: {}
};
exports["default"] = ResultComponent;
