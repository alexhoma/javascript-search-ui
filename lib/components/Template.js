"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var preact_1 = require("preact");
var Hogan = require("hogan.js");
/**
 * Template
 */
var Template = /** @class */ (function (_super) {
    __extends(Template, _super);
    function Template() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        /**
         * Render template
         *
         * @param template
         * @param result
         *
         * @return {any}
         */
        _this.renderTemplate = function (template, result) {
            /**
             * Compile template using hogan.js
             */
            var compiledTemplate = Hogan.compile(template);
            var output = compiledTemplate.render(result);
            return {
                __html: output
            };
        };
        return _this;
    }
    /**
     * Render
     *
     * @return {any}
     */
    Template.prototype.render = function () {
        var props = this.props;
        var template = props.template;
        var data = props.data;
        var className = props.className;
        return (template)
            ? preact_1.h("div", { className: className, dangerouslySetInnerHTML: this.renderTemplate(template, data) })
            : null;
    };
    return Template;
}(preact_1.Component));
exports["default"] = Template;
