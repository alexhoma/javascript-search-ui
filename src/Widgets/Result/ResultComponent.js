/** @jsx h */
import { h, Component } from 'preact';

class ResultComponent extends Component {
    constructor() {
        super();
    }

    render() {
        if (typeof this.props.data.items === 'undefined') {
            return null;
        }

        return (
            <ul>
                {this.props.data.items.map(item =>
                    <li>{item.uuid.id} - {item.uuid.type}</li>
                )}
            </ul>
        )
    }
}

export default ResultComponent;