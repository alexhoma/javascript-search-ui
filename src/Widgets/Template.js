/** @jsx h */
import { h, Component } from 'preact';
import Hogan from 'hogan.js';

class Template extends Component {
    renderTemplate = (template, data) => {
        let compiledTemplate = Hogan.compile(template);
        let output = compiledTemplate.render(data);


        return {
            __html: output
        };
    };

    render() {
        const {
            template,
            data,
            className
        } = this.props;

        return (template)
            ? <div
                className={className}
                dangerouslySetInnerHTML={this.renderTemplate(template, data)}
            />
            : null
        ;
    }
}

export default Template;