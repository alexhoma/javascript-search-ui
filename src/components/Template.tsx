import { h, Component } from 'preact';
import {TemplateProps} from "./TemplateProps";
import * as Mustache from 'mustache';

/**
 * Template
 */
class Template extends Component<TemplateProps> {

    /**
     * Render template
     *
     * @param template
     * @param result
     * @param dictionary
     *
     * @return {any}
     */
    renderTemplate = (template, result, dictionary) => {

        let trans = function() {
            return function(text, render) {
                return dictionary[text] ?? text;
            }
        }

        let output = Mustache.render(template, {...result, ...{
            'trans': trans
        }});

        return {
            __html: output
        };
    };

    /**
     * Render
     *
     * @return {any}
     */
    render() {

        const props = this.props;
        const template = props.template;
        const data = props.data;
        const className = props.className;
        const dictionary = props.dictionary ?? {};

        return (template)
            ? <div
                className={className}
                dangerouslySetInnerHTML={this.renderTemplate(template, data, dictionary)}
            />
            : null
        ;
    }
}

export default Template;
