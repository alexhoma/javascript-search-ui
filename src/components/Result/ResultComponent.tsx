import { h, Component } from 'preact';
import Template from "../Template";
import {defaultItemsListTemplate} from "./defaultTemplates";
import {configureQuery} from "./ResultActions";
import {ResultProps} from "./ResultProps";
import {ItemUUID} from "apisearch/lib/Model/ItemUUID";
import container from "../../Container";
import {
    APISEARCH_UI,
    APISEARCH_CONFIG,
} from "../../Constants";

/**
 * Result Component
 */
class ResultComponent extends Component<ResultProps> {

    /**
     * Component will mount
     */
    componentWillMount() {

        const props = this.props;

        /**
         * Dispatch action
         */
        configureQuery(
            props.environmentId,
            props.currentQuery,
            props.itemsPerPage,
            props.highlightsEnabled,
            props.promote.map((itemUUID) => {
                return itemUUID instanceof ItemUUID
                    ? itemUUID
                    : ItemUUID.createFromArray(itemUUID)
            }),
            props.exclude.map((itemUUID) => {
                return itemUUID instanceof ItemUUID
                    ? itemUUID
                    : ItemUUID.createFromArray(itemUUID)
            }),
            props.fields,
            props.filter
        );
    }

    /**
     * Render
     *
     * @return {any}
     */
    render() {

        const props = this.props;
        const dirty = props.dirty;
        const containerClassName = props.classNames.container;
        const itemsListClassName = props.classNames.itemsList;
        const placeholderClassName = props.classNames.placeholder;
        const environmentId = props.environmentId;
        const config = container.get(`${APISEARCH_CONFIG}__${environmentId}`);
        const apisearchUI = container.get(`${APISEARCH_UI}__${environmentId}`);
        const apisearchReference = apisearchUI.reference;

        const itemsListTemplate = props.template.itemsList;
        const placeholderTemplate = props.template.placeholder;

        const formatData = props.formatData;
        const currentResult = props.currentResult;
        const currentQuery = props.currentQuery;
        const currentVisibleResults = props.currentVisibleResults;

        if (!currentVisibleResults) {
            return (
                <div className={`as-result ${containerClassName}`}></div>
            )
        }

        /**
         * Data accessible to the template
         */
        const items = currentResult.getItems();
        let reducedTemplateData = {
            query: currentQuery.getQueryText()
        };

        /**
         * Format each item data
         */
        let formattedTemplateData = {
            ...reducedTemplateData,
            items: (items)
                ? items.map(function(item) {
                    let appId = config.app_id;
                    const appUUID = item.getAppUUID();
                    if (typeof appUUID === "object") {
                        appId = appUUID.composedUUID();
                    }

                    let indexId = config.index_id;
                    const indexUUID = item.getIndexUUID();
                    if (typeof indexUUID === "object") {
                        indexId = indexUUID.composedUUID();
                    }

                    const itemId = item.getUUID().composedUUID();
                    const userId = config.user_id;
                    const clickParameters = typeof userId === "string"
                        ? appId+'", "'+indexId+'", "'+itemId+'", "'+userId
                        : appId+'", "'+indexId+'", "'+itemId;

                    return {
                        ...formatData(item),
                        ...{
                            'click': apisearchReference + '.click("'+clickParameters+'");'
                        }
                    }
                })
                : [],
        };

        return (
            <div className={`as-result ${containerClassName}`}>
                {(placeholderTemplate && dirty)
                    ? <Template
                        template={placeholderTemplate}
                        className={`as-result__placeholder ${placeholderClassName}`}
                    />
                    : <Template
                        template={itemsListTemplate}
                        data={formattedTemplateData}
                        className={`as-result__itemsList ${itemsListClassName}`}
                    />
                }
            </div>
        )
    }
}

ResultComponent.defaultProps = {
    fields: [],
    itemsPerPage: 10,
    highlightsEnabled: false,
    promote: [],
    exclude: [],
    filter: function (query) {},
    classNames: {
        container: '',
        itemsList: '',
        placeholder: ''
    },
    template: {
        itemsList: defaultItemsListTemplate,
        placeholder: null
    },
    formatData: data => data
};

export default ResultComponent;
