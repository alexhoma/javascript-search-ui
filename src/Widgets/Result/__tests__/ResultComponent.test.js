/** @jsx h */
import {h} from 'preact';
import {deep, shallow} from 'preact-render-spy';
import ResultComponent from './../ResultComponent';
import Template from '../../Template';

describe('<ResultComponent />',() => {
    const defaultProps = {
        target: '#some-target',
        template: {
            itemsList: `
                <ul>
                {{#items}} 
                    <li>{{metadata.name}}</li> 
                {{/items}}
                </ul>
            `
        }
    };
    const data = {
        items: [
            {
                metadata: {
                    name: 'Tony Stark',
                    img: 'https://api.marvel.com/v2/characters/tony-stark'
                }
            },
            {
                metadata: {
                    name: 'Peter Parker',
                    img: 'https://api.marvel.com/v2/characters/peter-parker'
                }
            }
        ]
    };

    it('should render with default props', () => {
        ResultComponent.prototype.componentWillMount = jest.fn();
        const tree = deep(
            <ResultComponent
                {...defaultProps}
                data={data}
            />
        );

        /**
         * Testing the default props tree
         */
        expect(tree.component().props).toMatchSnapshot();

        /**
         * Testing rendered html
         */
        expect(tree).toMatchSnapshot();
    });
    it('should render with custom props', () => {
        ResultComponent.prototype.componentWillMount = jest.fn();
        const customProps = {
            itemsPerPage: 20,
            highlightsEnabled: true,
            classNames: {
                container: 'custom-container-class'
            }
        };

        const tree = deep(
            <ResultComponent
                {...defaultProps}
                {...customProps}
                data={data}
            />
        );

        /**
         * Testing the default props tree
         */
        expect(tree.component().props).toMatchSnapshot();
        /**
         * Testing rendered html
         */
        expect(tree).toMatchSnapshot();
    });
    it('should show placeholder if is set in custom properties and ApisearchUI state is dirty', () => {
        ResultComponent.prototype.componentWillMount = jest.fn();
        const customProps = {
            template: {
                placeholder: 'This is the results placeholder, search something!'
            }
        };

        const tree = deep(
            <ResultComponent
                dirty={true}
                {...defaultProps}
                {...customProps}
            />
        );

        expect(tree).toMatchSnapshot();
    })
    it('should update content when passing new data', () => {
        const tree = deep(
            <ResultComponent
                {...defaultProps}
                data={data}
            />
        );

        const updatedDataProps = {
            items: [
                {
                    metadata: {
                        name: 'Robert Bruce Banner',
                        img: 'https://api.marvel.com/v2/characters/robert-bruce-banner'
                    }
                },
                {
                    metadata: {
                        name: 'Thor Odinson',
                        img: 'https://api.marvel.com/v2/characters/thor-odinson'
                    }
                }
            ]
        }
        tree.render(
            <ResultComponent
                {...defaultProps}
                data={updatedDataProps}
            />
        )

        expect(tree).toMatchSnapshot();
    });
});