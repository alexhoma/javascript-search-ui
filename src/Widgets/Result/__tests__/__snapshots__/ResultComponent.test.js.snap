// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultComponent /> should render with custom props 1`] = `
Object {
  "children": Array [],
  "classNames": Object {
    "container": "custom-container-class",
  },
  "data": Object {
    "items": Array [
      Object {
        "metadata": Object {
          "img": "https://api.marvel.com/v2/characters/tony-stark",
          "name": "Tony Stark",
        },
      },
      Object {
        "metadata": Object {
          "img": "https://api.marvel.com/v2/characters/peter-parker",
          "name": "Peter Parker",
        },
      },
    ],
  },
  "itemsPerPage": 20,
  "target": "#some-target",
  "template": Object {
    "itemsList": "
                <ul>
                {{#items}} 
                    <li>{{metadata.name}}</li> 
                {{/items}}
                </ul>
            ",
  },
}
`;

exports[`<ResultComponent /> should render with custom props 2`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-result custom-container-class">
  <div class="asui-result--itemsList">
    
                    <ul>
                        <li>Tony Stark</li> 
                        <li>Peter Parker</li> 
                    </ul>
                
  </div>
</div>

`;

exports[`<ResultComponent /> should render with default props 1`] = `
Object {
  "children": Array [],
  "classNames": Object {
    "container": "",
  },
  "data": Object {
    "items": Array [
      Object {
        "metadata": Object {
          "img": "https://api.marvel.com/v2/characters/tony-stark",
          "name": "Tony Stark",
        },
      },
      Object {
        "metadata": Object {
          "img": "https://api.marvel.com/v2/characters/peter-parker",
          "name": "Peter Parker",
        },
      },
    ],
  },
  "itemsPerPage": 10,
  "target": "#some-target",
  "template": Object {
    "itemsList": "
                <ul>
                {{#items}} 
                    <li>{{metadata.name}}</li> 
                {{/items}}
                </ul>
            ",
  },
}
`;

exports[`<ResultComponent /> should render with default props 2`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-result ">
  <div class="asui-result--itemsList">
    
                    <ul>
                        <li>Tony Stark</li> 
                        <li>Peter Parker</li> 
                    </ul>
                
  </div>
</div>

`;

exports[`<ResultComponent /> should update content when passing new data 1`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-result ">
  <div class="asui-result--itemsList">
    
                    <ul>
                        <li>Robert Bruce Banner</li> 
                        <li>Thor Odinson</li> 
                    </ul>
                
  </div>
</div>

`;
